src/sss_client/ssh/sss_ssh_knownhosts.c

```diff
@@ -29,33 +29,92 @@
 #include "sss_client/sss_cli.h"
 #include "sss_client/ssh/sss_ssh_client.h"
 
+
+/*
+ * Parse the received hostname, which is expected in the format described in
+ * the “SSH_KNOWN_HOSTS FILE FORMAT” section of sshd(8). The parsed host name
+ * and port are returned as strings allocated with malloc(3), and not talloc(3),
+ * and must be freed by the caller.
+ *
+ * Some of the recognized formats are not expected from ssh, but it is easier
+ * to identify them and useful in the case the tool is launched manually by a
+ * user.
+ *
+ * If any of the expected values (host or port) is not found, their respective
+ * output arguments will NOT be modified.
+ */
+static errno_t parse_ssh_host(const char *ssh_host,
+                              const char **_host, const char **_port)
+{
+    int values;
+
+    /* Host name between brackets and with a port number.
+     * ssh can use this format.
+     */
+    values = sscanf(ssh_host, "[%m[^]]]:%ms", _host, _port);
+    if (values == 2) {
+        return EOK;
+    }
+    /* Just a host name enclosed between brackets.
+     * ssh is not expected to use this format but... who knows?
```

Comment:
in general using brackets like this have been introduced with IPv6 to clearly distinguish ":" port separator.

Could you update the comment like:
SSH may use this format when the "hostname" is pure IPv6 address...

Reply:
Actually `ssh` uses it also used for IPv4.